src/Bicep.Types/Concrete/ResourceType.cs

```diff
@@ -15,8 +16,8 @@ public enum ResourceFlags
     public class ResourceType : TypeBase
     {
         [JsonConstructor]
-        public ResourceType(string name, ScopeType scopeType, ScopeType? readOnlyScopes, ITypeReference body, ResourceFlags flags)
-            => (Name, ScopeType, ReadOnlyScopes, Body, Flags) = (name, scopeType, readOnlyScopes, body, flags);
+        public ResourceType(string name, ScopeType scopeType, ScopeType? readOnlyScopes, ITypeReference body, ResourceFlags flags, IReadOnlyList<ITypeReference>? functions)
```

Comment:
Just checking my work, it looks like based on `IReadOnlyList<ITypeReference>? functions` The function name is part of the function's type signature, does that sound right? It sounds strange to me.